java/json/src/main/java/com/microsoft/azure/schemaregistry/kafka/json/KafkaJsonDeserializer.java

```diff
@@ -0,0 +1,118 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+package com.microsoft.azure.schemaregistry.kafka.json;
+
+import java.util.Arrays;
+import java.util.Map;
+import java.util.Set;
+import org.apache.kafka.common.header.Headers;
+import org.apache.kafka.common.serialization.Deserializer;
+import com.azure.data.schemaregistry.SchemaRegistryClient;
+import com.azure.data.schemaregistry.SchemaRegistryClientBuilder;
+import com.azure.data.schemaregistry.models.SchemaRegistrySchema;
+import com.fasterxml.jackson.annotation.JsonAutoDetect;
+import com.fasterxml.jackson.databind.DeserializationFeature;
+import com.fasterxml.jackson.databind.JsonNode;
+import com.fasterxml.jackson.databind.ObjectMapper;
+import com.networknt.schema.JsonSchema;
+import com.networknt.schema.JsonSchemaFactory;
+import com.networknt.schema.SpecVersion;
+import com.networknt.schema.ValidationMessage;
+
+/**
+ * Deserializer implementation for Kafka consumer, implementing Kafka Deserializer interface.
+ *
+ * @see KafkaJsonSerializer See serializer class for upstream serializer implementation
+ */
+public class KafkaJsonDeserializer<T> implements Deserializer<T> {
+    private SchemaRegistryClient client;
+    private KafkaJsonDeserializerConfig config;
+
+    /**
+     * Empty constructor used by Kafka consumer
+     */
+    public KafkaJsonDeserializer() {
+        super();
+    }
+
+    /**
+     * Configures deserializer instance.
+     *
+     * @param props Map of properties used to configure instance
+     * @param isKey Indicates if deserializing record key or value.  Required by Kafka deserializer interface,
+     *              no specific functionality has been implemented for key use.
+     *
+     * @see KafkaJsonDeserializerConfig Deserializer will use configs found in here and inherited classes.
+     */
+    public void configure(Map<String, ?> props, boolean isKey) {
+        this.config = new KafkaJsonDeserializerConfig((Map<String, Object>) props);
+
+        this.client = new SchemaRegistryClientBuilder()
+        .fullyQualifiedNamespace(this.config.getSchemaRegistryUrl())
+        .credential(this.config.getCredential())
```

Comment:
Per discussion with SDK team, we should add a Kafka specific application id here so that service side can trace the client telemetry.

```
        this.client = new SchemaRegistryClientBuilder()
        .fullyQualifiedNamespace(this.config.getSchemaRegistryUrl())
        .credential(this.config.getCredential())
        .clientOptions(new ClientOptions().setApplicationId("KafKaSerializer/[some_version]"))
        .buildClient();
```

Reply:
and that also means we need to add Telemetry tracking for this application ID on the SR service side.